DSharpPlus/Entities/AutoModeration/RuleTriggerMetadata.cs

```diff
@@ -0,0 +1,67 @@
+using DSharpPlus.Enums;
+
+using Newtonsoft.Json;
+
+namespace DSharpPlus.Entities;
+
+public sealed class RuleTriggerMetadata
+{
+    [JsonProperty("keyword_filter")]
+    public string[]? KeywordFilter { get; internal set; }
+
+    [JsonProperty("regex_patterns")]
+    public string[]? RegexPatterns { get; internal set; }
+
+    [JsonProperty("presets")]
+    public RuleKeywordPresetType[] KeywordPresetTypes { get; internal set; }
+
+    [JsonProperty("allow_list")]
+    public string[] AllowedKeywords { get; internal set; }
+
+    [JsonProperty("mention_total_limit")]
+    public int? MentionTotalLimit { get; internal set; }
+
+    internal RuleTriggerMetadata()
+    {
+
+    }
+}
+
+public sealed class RuleTriggerMetadataBuilder
+{
+    public string[]? KeywordFilter { get; set; }
+
+    public string[]? RegexPatterns { get; set; }
+
+    public RuleKeywordPresetType[] KeywordPresetTypes { get; set; }
+
+    public string[] AllowedKeywords { get; set; }
+
+    public int? MentionTotalLimit { get; set; }
+
+    public RuleTriggerMetadataBuilder()
+    {
+
+    }
+
+    public RuleTriggerMetadataBuilder(string[]? keywordFilter, string[]? regexPatterns, RuleKeywordPresetType[] keywordPresetTypes, string[] allowKeywords, int? mentionTotalLimit)
```

Comment:
this probably does deserve to have a builder type, but it should implement the builder properly (see [https://github.com/DSharpPlus/DSharpPlus/blob/5ae73cd825ec864be03bff213b48dc779dabaec3/DSharpPlus/Entities/Channel/Message/DiscordMessageBuilder.cs#L147](https://github.com/DSharpPlus/DSharpPlus/blob/5ae73cd825ec864be03bff213b48dc779dabaec3/DSharpPlus/Entities/Channel/Message/DiscordMessageBuilder.cs#L147)).